docs/source/extension/extension_points.rst

```diff
@@ -721,6 +723,117 @@ When the ``labStatus`` busy state changes, we update the text content of the
+       | undefined;
+   
+     // Register notebook toolbar widgets
+     if (toolbarRegistry) {
+       toolbarRegistry.registerFactory<NotebookPanel>(FACTORY, 'cellType', panel =>
+         ToolbarItems.createCellTypeItem(panel, translator)
+       );
+       
+       toolbarRegistry.registerFactory<NotebookPanel>(
+         FACTORY,
+         'kernelStatus',
+         panel => Toolbar.createKernelStatusItem(panel.sessionContext, translator)
+       );
+       // etc... 
+     
+       if (settingRegistry) {
+         // Create the factory
+         toolbarFactory = createToolbarFactory(
+           toolbarRegistry,
+           settingRegistry,
+           // Factory name
+           FACTORY,
+           // Setting id in which the toolbar items are defined
+           '@jupyterlab/notebook-extension:panel',
+           translator
+         );
+       }
+     }
+   
+     const factory = new NotebookWidgetFactory({
+       name: FACTORY,
+       fileTypes: ['notebook'],
+       modelName: 'notebook',
+       defaultFor: ['notebook'],
+       // ...
+       toolbarFactory,
+       translator: translator
+     });
+     app.docRegistry.addWidgetFactory(factory);
+
+The registry ``registerFactory`` method allows an extension to provide special widget for a unique pair 
+(factory name, toolbar item name). Then the helper ``createToolbarFactory`` can be used to extract the
+toolbar definition from the settings and build the factory to pass to the widget factory.
+
+The default toolbar items can be defined across multiple extensions by providing an entry in the ``"jupyter.lab.toolbars"``
+mapping. For example for the html viewer:
+
+.. code:: json
+ 
+   "jupyter.lab.toolbars": {
+     "Notebook": [
+       { "name": "save", "rank": 10 },
+       { "name": "insert", "command": "notebook:insert-cell-below", "rank": 20 },
+       { "name": "cut", "command": "notebook:cut-cell", "rank": 21 },
+       { "name": "copy", "command": "notebook:copy-cell", "rank": 22 },
+       { "name": "paste", "command": "notebook:paste-cell-below", "rank": 23 },
+       { "name": "run", "command": "runmenu:run", "rank": 30 },
+       { "name": "interrupt", "command": "kernelmenu:interrupt", "rank": 31 },
+       { "name": "restart", "command": "kernelmenu:restart", "rank": 32 },
+       {
+         "name": "restart-and-run",
+         "command": "runmenu:restart-and-run-all",
+         "rank": 33
+       },
+       { "name": "cellType", "rank": 40 },
+       { "name": "spacer", "type": "spacer", "rank": 100 },
+       { "name": "kernelName", "rank": 1000 },
+       { "name": "kernelStatus", "rank": 1001 }
+     ]
+   },
+   "jupyter.lab.transform": true,
+   "properties": {
+     "toolbar": {
+       "title": "Notebook panel toolbar items",
+       "items": {
+         "$ref": "#/definitions/toolbarItem"
+       },
+       "type": "array",
+       "default": []
+     }
+   }
+
+The settings registry will gather those definition within the provided setting plugin id
```

Comment:
```suggestion
The settings registry will gather those definitions within the provided setting plugin id
```

Reply:
I am having difficulty understanding what `setting plugin id` refers to in this and following sentences.

Reply:
This is indeed a not easy explanation. 

As for the shortcuts any extension can define a `jupyter.lab.toolbars`, the final aggregated list is built and set as default value on a `toolbar` property in a setting plugin example `@jupyterlab/notebook-extension:panel`. This then allow the user to override the toolbar definition. 

If you find a better way to explain this, that would be great.

Reply:
Ok, I get it now so it refers to the ID passed to `createToolbarFactory`.